light-clients/ics10-grandpa/src/client_message.rs

```diff
@@ -44,6 +47,14 @@ pub struct Header {
 	/// finalzed at the relay chain height. We check for this parachain header finalization
 	/// via state proofs. Also contains extrinsic proof for timestamp.
 	pub parachain_headers: BTreeMap<H256, ParachainHeaderProofs>,
+	/// Lazily initialized height
+	pub height: Height,
```

Comment:
Why do we need this?

Reply:
This is because the wasm client may arbitrarily request the height, so we need to store it somewhere. Otherwise, height calculation becomes very resource-intensive.